### PR TITLE
Evaluated model improvements

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Various improvements of the evaluated model, see the commit messages for details.

### CODE REVIEW

File: reporter/src/main/kotlin/model/EvaluatedModelMapper.kt [hunk 1]
```diff
@@ -162,6 +163,15 @@ internal class EvaluatedModelMapper(private val input: ReporterInput) {
         }
     }
 
+    private fun createScopes() {
+        input.ortResult.getProjects().flatMap { project -> project.scopes.map { it.name } }.forEach { scope ->
```
Comment (sschuberth) [inline reporter/src/main/kotlin/model/EvaluatedModelMapper.kt:168]: Nit: How about expanding the formatting a bit like

        input.ortResult.getProjects().flatMap { project ->
            project.scopes.map { it.name }
        }.forEach { scope ->
            scopes[scope] = EvaluatedScope(
                name = scope,
                excludes = scopeExcludes.addIfRequired(input.ortResult.getExcludes().findScopeExcludes(scope))
            )
        }

File: reporter/src/funTest/assets/static-html-reporter-test-expected-output.html [hunk 2]
```diff
@@ -424,7 +424,7 @@
 </script>
     <div id="report-container">
       <div class="ort-report-label">Scan Report</div>
-      <div>Created by <strong>ORT</strong>, the <a href="http://oss-review-toolkit.org/">OSS Review Toolkit</a>, version <REPLACE_ORT_VERSION> on <REPLACE_TIMESTAMP>.</div>
+      <div>Created by <strong>ORT</strong>, the <a href="http://oss-review-toolkit.org/">OSS Review Toolkit</a>, version 0.1.0-SNAPSHOT on <REPLACE_TIMESTAMP>.</div>
```
Comment (sschuberth) [inline reporter/src/funTest/assets/static-html-reporter-test-expected-output.html:427]: This needs to stay `<REPLACE_ORT_VERSION>`.

File: reporter/src/main/kotlin/model/EvaluatedModel.kt [hunk 3]
```diff
@@ -108,8 +109,11 @@ data class EvaluatedModel(
     val ruleViolationResolutions: List<RuleViolationResolution>,
     val ruleViolations: List<EvaluatedRuleViolation>,
     val statistics: Statistics,
-    // TODO: Ideally this would be an instance of RepositoryConfiguration, but for now it has to be a string to not be
-    //       converted to JSON when using it as input for the web app reporter.
+    val repository: Repository,
+    /**
```
Comment (sschuberth) [inline reporter/src/main/kotlin/model/EvaluatedModel.kt:113]: Nit: Now that we have KDoc, we usually group it to the property by adding a blank line before this one, and a blank line after the property.

File: reporter/src/main/kotlin/model/EvaluatedModelMapper.kt (hunk 1 above)
Comment (mnonnenmacher) [reply reporter/src/main/kotlin/model/EvaluatedModelMapper.kt]: Done.

File: reporter/src/funTest/assets/static-html-reporter-test-expected-output.html (hunk 2 above)
Comment (mnonnenmacher) [reply reporter/src/funTest/assets/static-html-reporter-test-expected-output.html]: Done.

File: reporter/src/main/kotlin/model/EvaluatedModel.kt (hunk 3 above)
Comment (mnonnenmacher) [reply reporter/src/main/kotlin/model/EvaluatedModel.kt:114]: Done.

File: reporter/src/main/kotlin/model/EvaluatedModelMapper.kt [hunk 4]
```diff
@@ -162,6 +164,17 @@ internal class EvaluatedModelMapper(private val input: ReporterInput) {
         }
     }
 
+    private fun createScopes() {
+        input.ortResult.getProjects().flatMap {
+                project -> project.scopes.map { it.name }
```
Comment (sschuberth) [inline reporter/src/main/kotlin/model/EvaluatedModelMapper.kt:169]: Wrong formatting: `project ->` belongs to the previous line.

Comment (mnonnenmacher) [reply reporter/src/main/kotlin/model/EvaluatedModelMapper.kt]: Done.

**Pull request merged**